### PR TITLE
Use pip-compile for tracking dependencies

### LINKED ISSUE #689: tests fail with numpy 2.0
```
[task 2024-06-19T13:43:12.117Z] ==================================== ERRORS ====================================
[task 2024-06-19T13:43:12.117Z] _________________ ERROR collecting tests/test_tracking_cli.py __________________
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/runner.py:341: in from_call
[task 2024-06-19T13:43:12.117Z]     result: Optional[TResult] = func()
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/runner.py:372: in <lambda>
[task 2024-06-19T13:43:12.117Z]     call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/python.py:531: in collect
[task 2024-06-19T13:43:12.117Z]     self._inject_setup_module_fixture()
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
[task 2024-06-19T13:43:12.117Z]     self.obj, ("setUpModule", "setup_module")
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/python.py:310: in obj
[task 2024-06-19T13:43:12.117Z]     self._obj = obj = self._getobj()
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/python.py:528: in _getobj
[task 2024-06-19T13:43:12.117Z]     return self._importtestmodule()
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/python.py:617: in _importtestmodule
[task 2024-06-19T13:43:12.117Z]     mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/pathlib.py:567: in import_path
[task 2024-06-19T13:43:12.117Z]     importlib.import_module(module_name)
[task 2024-06-19T13:43:12.117Z] /usr/lib/python3.10/importlib/__init__.py:126: in import_module
[task 2024-06-19T13:43:12.117Z]     return _bootstrap._gcd_import(name[level:], package, level)
[task 2024-06-19T13:43:12.117Z] <frozen importlib._bootstrap>:1050: in _gcd_import
[task 2024-06-19T13:43:12.117Z]     ???
[task 2024-06-19T13:43:12.117Z] <frozen importlib._bootstrap>:1027: in _find_and_load
[task 2024-06-19T13:43:12.117Z]     ???
[task 2024-06-19T13:43:12.117Z] <frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
[task 2024-06-19T13:43:12.117Z]     ???
[task 2024-06-19T13:43:12.117Z] <frozen importlib._bootstrap>:688: in _load_unlocked
[task 2024-06-19T13:43:12.117Z]     ???
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
[task 2024-06-19T13:43:12.117Z]     exec(co, module.__dict__)
[task 2024-06-19T13:43:12.117Z] tests/test_tracking_cli.py:10: in <module>
[task 2024-06-19T13:43:12.117Z]     from translations_parser.cli import experiments as experiments_publish
[task 2024-06-19T13:43:12.117Z] tracking/translations_parser/cli/experiments.py:16: in <module>
[task 2024-06-19T13:43:12.117Z]     from translations_parser.parser import TrainingParser
[task 2024-06-19T13:43:12.117Z] tracking/translations_parser/parser.py:13: in <module>
[task 2024-06-19T13:43:12.117Z]     from translations_parser.publishers import Publisher
[task 2024-06-19T13:43:12.117Z] tracking/translations_parser/publishers.py:9: in <module>
[task 2024-06-19T13:43:12.117Z]     import wandb
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/wandb/__init__.py:27: in <module>
[task 2024-06-19T13:43:12.117Z]     from wandb import sdk as wandb_sdk
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/wandb/sdk/__init__.py:4: in <module>
[task 2024-06-19T13:43:12.117Z]     from .artifacts.artifact import Artifact  # noqa: F401
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/wandb/sdk/artifacts/artifact.py:35: in <module>
[task 2024-06-19T13:43:12.117Z]     from wandb import data_types, env, util
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/wandb/data_types.py:33: in <module>
[task 2024-06-19T13:43:12.117Z]     from .sdk.data_types import _dtypes
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/wandb/sdk/data_types/_dtypes.py:396: in <module>
[task 2024-06-19T13:43:12.117Z]     NumberType.types.append(np.float_)
[task 2024-06-19T13:43:12.117Z] ../../.cache/pypoetry/virtualenvs/firefox-translations-training-hlUNdH3O-py3.10/lib/python3.10/site-packages/numpy/__init__.py:397: in __getattr__
[task 2024-06-19T13:43:12.117Z]     raise AttributeError(
[task 2024-06-19T13:43:12.117Z] E   AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?
[task 2024-06-19T13:43:12.117Z] =========================== short test summary info ============================
[task 2024-06-19T13:43:12.117Z] ERROR tests/test_tracking_cli.py - AttributeError: `np.float_` was removed in...
[task 2024-06-19T13:43:12.117Z] !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
[task 2024-06-19T13:43:12.117Z] =============================== 1 error in 1.91s ===============================
```

### PR DESCRIPTION
**Pull request by La0**
Closes #689

### CODE REVIEW

File: tracking/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile requirements.in
+#
+aiohttp==3.9.5
```
[the remark this answers is not in the document]
Comment (La0) [reply tracking/requirements.txt]: Done !

**Comment by La0** [comment]
@bhearsum `pip` with `pyproject.toml` does not seem to parse correctly the `requirements.txt` file with hashes (removing all white spaces on each line).
Have you ever encountered that ?

**Comment by bhearsum** [comment]
> @bhearsum `pip` with `pyproject.toml` does not seem to parse correctly the `requirements.txt` file with hashes (removing all white spaces on each line). Have you ever encountered that ?

That's new to me, but I've also never used a requirements file reference in pyproject.toml. Everywhere else we do this we `pip install` from a requirements file.

**Comment by bhearsum** [comment]
I think I'm very confused though. Dependencies are being installed with `poetry install`. Doesn't it (shouldn't it?) be sourcing dependencies from a Poetry lockfile instead?

**Comment by La0** [comment]
AFAIK the tracking code is installed through `pip install` in the tasks: https://github.com/mozilla/firefox-translations-training/blob/main/taskcluster/kinds/train-teacher/kind.yml#L125 (same for the other dependencies).

I initially did not include the hashes, as the file [pointed out](https://github.com/mozilla/firefox-translations-training/issues/689#issuecomment-2181255501) by Evgeny [did not have those](https://github.com/mozilla/firefox-translations-training/tree/main/pipeline/clean/requirements)

**Comment by La0** [comment]
@eu9ene I have reverted to my initial patch, without hashes, that works on CI.

**Review by eu9ene — APPROVED** [review]
Let's merge if it fixes the problem as it's blocking all other PRs. @bhearsum I don't think we've been using the approach with hashes much, but it's definitely something we should consider.

@La0 I hope this will work with our `pip3 install $VCS_PATH/tracking &&` in Taskcluster kinds, but ideally we should run the CI with publishing to retest that.

**Comment by eu9ene** [comment]
I tested it here and it looks working: https://firefox-ci-tc.services.mozilla.com/tasks/groups/GdnVgLB1Q7Kcjqoej9mEOA

**Pull request merged**